File: twoSum (easy)/twoSum.js
function twoSum(nums, target) {

    const hashTable = {} // Creating a hash table manually rather than a Map()
    
    for (let i = 0; i<nums.length; i++) { // Iterating
     const currentNumber = nums[i]; // Storing the current number inside the array
     
     if (hashTable[target - currentNumber] !== undefined) { // If the difference between currentNumber and the target has been found previously in the loop...
        return [hashTable[target - currentNumber], i]; // Return the indices from hashTable
     }
     
     hashTable[currentNumber] = i // Store the currentNumber as a key with the index as the number
    
    }
    
    }

File: twoSum (easy)/twoSum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        hashTable = {}  # Create a hash table

        for i, currentNumber in enumerate(nums):  # Iterating with index and value - lets use enumerate instead of a counter, too.
            if target - currentNumber in hashTable:  # If we hit bingo in hashTable...
                return [hashTable[target - currentNumber], i]  # ...return the indices!

            hashTable[currentNumber] = i  # Store the current iterated number with its index and move on

        return []  # Return empty list if there's no valid solution